app/models/manager_refresh/inventory_collection.rb

```diff
@@ -947,6 +963,19 @@ def full_collection_for_comparison
 
     attr_writer :attributes_blacklist, :attributes_whitelist, :db_data_index, :parent_inventory_collections
 
+    # Returns array of records identities
+    def records_identities(records)
+      records = [records] unless records.kind_of?(Array)
+      records.map { |record| record_identity(record) }
+    end
+
+    # Returns a hash with a simple record identity
+    def record_identity(record)
+      {
+        :id => record.try(:[], :id) || record.try(:id)
```

Comment:
hm, it should not fail. The storing is only on places with actual created/updated/deleted records.

But I can add the exception.